packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -40,6 +69,49 @@ function UnconnectedNavigatorProvider(
 			path: initialPath,
 		},
 	] );
+	const [ screens, dispatch ] = useReducer( screensReducer, [] );
+	const [ matchedPath, setMatchedPath ] = useState< MatchedPath >( false );
+	const currentMatch = useRef< MatchedPath >( false );
+
+	useEffect( () => {
+		const updateMatch = ( newMatch: MatchedPath ) => {
+			setMatchedPath( newMatch );
+			currentMatch.current = newMatch;
+		};
+
+		const resolvePath = ( path: string ) => {
+			const newMatch = patternMatch( path, screens );
+			if (
+				currentMatch.current !== false &&
+				newMatch !== false &&
+				isShallowEqual( newMatch, currentMatch.current )
+			) {
+				return;
+			}
+
+			updateMatch( newMatch );
+		};
+
+		if ( locationHistory.length > 0 ) {
+			const path = locationHistory[ locationHistory.length - 1 ].path;
+			if ( path !== undefined ) {
+				resolvePath( path );
+			} else {
+				updateMatch( false );
+			}
+		}
+	}, [ screens, locationHistory ] );
```

Comment:
Actually this one should be a `useMemo` and not a `useEffect` :) I'll make that update soon.